### DISCUSSION TITLE
How to open MenuItem in new tab

### DISCUSSION THREAD

**Discussion by khanakia** [filamentphp/filament#8347] [Help]
I am using this example to open menu in new tab but i could not find any option
https://filamentphp.com/docs/3.x/panels/navigation#customizing-the-user-menu
```php
use App\Filament\Pages\Settings;
use Filament\Navigation\MenuItem;
use Filament\Panel;
 
public function panel(Panel $panel): Panel
{
    return $panel
        // ...
        ->userMenuItems([
            MenuItem::make()
                ->label('Settings')
                ->url(fn (): string => Settings::getUrl())
                ->icon('heroicon-o-cog-6-tooth'),
            // ...
        ]);
}

```

**Comment by bcollins-oxford** [comment]
Typically a ->openUrlInNewTab(true) would suffice with a URL, however, I'm not sure with MenuItem.

**Reply by khanakia** [reply]
Getting error `Method Filament\Navigation\MenuItem::openUrlInNewTab does not exist.`

**Reply by leandrocfe** [reply]
Maybe you can use `panels::user-menu.profile.after` [render-hook](https://filamentphp.com/docs/3.x/support/render-hooks#available-render-hooks) to render a custom link or component

**Reply by khanakia** [reply]
I can try but I am not sure why we do not have any method on `MenuItem::make()` as this is the most common use case and many users could need this.

**Reply by danharrin** [reply]
You can quite easily PR openUrlInNewTab() in, just copy it from other files

**Reply by khanakia** [reply]
I have created a pull request can check this please https://github.com/filamentphp/filament/pull/8379
